test_cli_remapping/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(test_cli_remapping)
+
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
+
+if(BUILD_TESTING)
+  find_package(ament_cmake REQUIRED)
+  find_package(ament_lint_auto REQUIRED)
+  find_package(ament_cmake_pytest REQUIRED)
+
+  ament_lint_auto_find_test_dependencies()
+
+  add_executable(name_maker_rclcpp
+    test/name_maker.cpp)
+  target_compile_features(name_maker_rclcpp PRIVATE cxx_lambdas)
```

Comment:
Something like it is. Alternatively I can change this to `set_property(TARGET name_maker_rclcpp PROPERTY CXX_STANDARD 11)`

Reply:
Would setting the the CMAKE_CXX_STANDARD globally like we do in the other packages do the job ?
https://github.com/ros2/system_tests/blob/84c035d4e11ad99f5ac19449fcb9ed5b11d91c01/test_communication/CMakeLists.txt#L5-L8

Reply:
It would, though I'm not a fan of setting things globally in cmake. I find it easier to debug stuff when I can see what is set on a specific target.

Reply:
> I'm not a fan of setting things globally in cmake

In general I agree.
In this specific case, we do [target and rely on c++14](https://github.com/ros2/ros2/wiki/Developer-Guide#c-1) globally in ROS 2 so I think that setting the standard globally should be done in all packages to avoid inconsistency or surprises.

Reply:
> In this specific case, we do target and rely on c++14 globally in ROS 2 so I think that setting the standard globally should be done in all packages to avoid inconsistency or surprises.

Thats a good point for this package. I'll change it.

For downstream users I disagree about it meaning the standard should be set globally. Instead I think `ament_target_depenencies()` below should have given the target the standard required to include `rclcpp` headers. This would act on a package level like  `target_compile_features(... PUBLIC ...)` does on a target level.

Reply:
0c87b6a